include/grpcpp/impl/codegen/async_unary_call.h

```diff
@@ -134,61 +244,36 @@ class ClientAsyncResponseReader final
   ///   - the \a ClientContext associated with this call is updated with
   ///     possible initial and trailing metadata sent from the server.
   void Finish(R* msg, ::grpc::Status* status, void* tag) override {
-    GPR_CODEGEN_ASSERT(started_);
-    if (initial_metadata_read_) {
-      finish_buf.set_output_tag(tag);
-      finish_buf.RecvMessage(msg);
-      finish_buf.AllowNoMessage();
-      finish_buf.ClientRecvStatus(context_, status);
-      call_.PerformOps(&finish_buf);
-    } else {
-      single_buf.set_output_tag(tag);
-      single_buf.RecvInitialMetadata(context_);
-      single_buf.RecvMessage(msg);
-      single_buf.AllowNoMessage();
-      single_buf.ClientRecvStatus(context_, status);
-      call_.PerformOps(&single_buf);
-    }
+    GPR_CODEGEN_DEBUG_ASSERT(started_);
+    finish_(context_, &call_, initial_metadata_read_, single_buf_, &finish_buf_,
+            static_cast<void*>(msg), status, tag);
   }
 
  private:
-  friend class internal::ClientAsyncResponseReaderFactory<R>;
+  friend class internal::ClientAsyncResponseReaderHelper;
   ::grpc::ClientContext* const context_;
   ::grpc::internal::Call call_;
-  bool started_;
+  bool started_ = false;
```

Comment:
Is this a bug fix BTW?

Reply:
No. It used to be initialized from a parameter to the constructor, but the new constructor no longer has that parameter.